From: Jason Rhinelander <jason@imaginary.ca>
Date: Wed, 26 Oct 2022 17:19:00 -0300
Subject: Fix time_delta<seconds> formatting in milliseconds

The time_delta<T> was using the wrong duration type when formatting, so
was outputting millisecond precision in the systemd status string which
is pointless (and unintended).
---
 llarp/util/time.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/llarp/util/time.hpp b/llarp/util/time.hpp
index fbf9997..c50c28b 100644
--- a/llarp/util/time.hpp
+++ b/llarp/util/time.hpp
@@ -42,7 +42,7 @@ namespace fmt
     format(const llarp::time_delta<Time_Duration>& td, FormatContext& ctx)
     {
       const auto dlt =
-          std::chrono::duration_cast<llarp::Duration_t>(llarp::TimePoint_t::clock::now() - td.at);
+          std::chrono::duration_cast<Time_Duration>(llarp::TimePoint_t::clock::now() - td.at);
       using Parent = formatter<std::string>;
       if (dlt > 0s)
         return Parent::format(fmt::format("{} ago", dlt), ctx);
